server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -320,33 +372,81 @@ private String limitLength(String description) {
             }
         }
 
+        public void publish() {
+            publisher.publish(
+                currentInfo,
+                ActionListener.wrap(
+                    r -> logger.debug("Successfully published health indicator"),
+                    e -> logger.warn("Failed to publish health indicator", e)
+                )
+            );
+        }
+
         @Override
         public String name() {
             return NAME;
         }
 
         @Override
         public synchronized HealthIndicatorResult calculate(boolean verbose, int maxAffectedResourcesCount, HealthInfo healthInfo) {
-            if (isActive == false) {
+            return calculate(healthInfo.fileSettingsHealthInfo());
+        }
+
+        public HealthIndicatorResult calculateFromCurrentInfo() {
+            return calculate(currentInfo);
+        }
```

Comment:
Is there a reason you chose not to extract a health "tracker" class (i.e. a class that actually holds the reference to `currentInfo` and has the mutation methods)? That was my suggestion in https://github.com/elastic/elasticsearch/pull/127397#discussion_r2063407877. I think that's still worth doing as it separates concerns and ensures that the health indicator does not have any state (which it shouldn't have).

That also allows us to remove the `calculateFromCurrentInfo`, which can lead to confusion IMO.

Reply:
No, not a real reason. There are a lot of moving parts here and I was mostly trying to get it working correctly first.

Reply:
Added `FileSettingsHealthTracker`.